src/legacy/core_plugins/kbn_vislib_vis_types/public/editors/metrics_axes_options.tsx

```diff
@@ -0,0 +1,316 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState, useEffect, useCallback } from 'react';
+import { cloneDeep, capitalize, get, uniq } from 'lodash';
+import { EuiSpacer } from '@elastic/eui';
+
+import { AggConfig } from 'ui/vis';
+import { AggGroupNames } from 'ui/vis/editors/default';
+import { safeMakeLabel } from 'ui/agg_types/agg_utils';
+import { BasicVislibParams, ValueAxis, SeriesParam, Axis } from '../types';
+import { ValidationVisOptionsProps } from '../controls/validation_wrapper';
+import { SeriesPanel } from '../controls/point_series/series_panel';
+import { CategoryAxisPanel } from '../controls/point_series/category_axis_panel';
+import { ValueAxesPanel } from '../controls/point_series/value_axes_panel';
+import { mapPositionOpposite, mapPosition } from '../controls/point_series/utils';
+import { makeSerie, isAxisHorizontal, countNextAxisNumber } from './metrics_axes_options_helper';
+
+export type SetChartValueByIndex = <T extends keyof SeriesParam>(
+  index: number,
+  paramName: T,
+  value: SeriesParam[T]
+) => void;
+
+export type SetValueAxisByIndex = <T extends keyof ValueAxis>(
+  index: number,
+  paramName: T,
+  value: ValueAxis[T]
+) => void;
+
+const VALUE_AXIS_PREFIX = 'ValueAxis-';
+const AXIS_PREFIX = 'Axis-';
+
+function MetricsAxisOptions(props: ValidationVisOptionsProps<BasicVislibParams>) {
+  const { stateParams, setValue, aggs, setVisType, vis } = props;
+
+  const [lastCustomLabels, setLastCustomLabels] = useState({} as { [key: string]: string });
+  const [isCategoryAxisHorizontal, setIsCategoryAxisHorizontal] = useState(true);
+  const [axesNumbers, setAxesNumbers] = useState({} as { [key: string]: number });
+  // We track these so we can know when the agg is changed
+  const [lastMatchingSeriesAggType, setLastMatchingSeriesAggType] = useState('');
+  const [lastMatchingSeriesAggField, setLastMatchingSeriesAggField] = useState('');
+
+  const setValueAxisByIndex: SetValueAxisByIndex = useCallback(
```

Comment:
Seems `setValueAxisByIndex` and `setChartByIndex` could be one function with additional key passed in